js/build.js

```diff
@@ -9,181 +14,153 @@ Fliplet.Widget.instance({
+        }
+
+        return typeof elem === 'string' ? 'string' : false;
+      }
+
+      function decodeHTMLEntities(str) {
+        var temp = document.createElement('div');
+
+        temp.style.display = 'none';
+        temp.innerHTML = str;
+
+        return temp.textContent || temp.innerText;
+      }
+
+      function getParsedType(elem) {
+        try {
+          let parsedValue = JSON.parse(elem);
+
+          if (Array.isArray(parsedValue)) {
+            return 'array';
+          }
+
+          return false;
+        } catch (error) {
+          return 'string';
+        }
+      }
+
+      function evaluate(condition, expression) {
+        try {
+          if (eval(expression)) {
+            return condition['visibility'];
+          }
+        } catch (error) {
+          return false;
+        }
+      }
+
+      function setResult(expr) {
+        result = expr && expr !== 'hide';
+      }
+
+      function isConditionIncluded(array, condition) {
+        const userValue = condition.user_value;
+
+        if (array.includes(userValue) || array.some(value => typeof value === 'number' && value === Number(userValue))) {
+          return setResult(condition.visibility);
+        }
+      }
+
       let helper = this;
       let conditions = this.fields.conditions;
-      let environment = Fliplet.Env.get('preview');
-      $(helper.el).addClass('hidden'); //by default button is hidden
-
-      return Fliplet.Session.get().then(function onSessionRetrieved(session) {
-        if (session && session.entries) {
-          if (session.entries.dataSource) {
-            let user = session.entries.dataSource.data;
-            let result;
-
-            function ifArray(elem) {
-              if (Array.isArray(elem)) {
-                return 'array';
-              } else if (typeof elem === 'string') {
-                return 'string';
-              } else { return false; }
-            }
+      let isPreview = Fliplet.Env.get('preview');
 
-            function ifValidJson(elem) {
-              try {
-                JSON.parse(elem);
-                let parsedValue = JSON.parse(elem);
-                if (Array.isArray(parsedValue)) {
-                  return 'array';
-                }
-                return false;
-              } catch (error) {
-                return 'string';
-              }
-            }
+      $(helper.el).addClass('hidden'); // by default button is hidden
 
-            function evaluate(condition, expression) {
-              try {
-                eval(expression);
-                if (eval(expression)) {
-                  return condition['visibility'];
-                }
-              } catch (error) {
-                console.log('Expression can\'t be evaluated. Error: ' + error);
-                return false;
-              }
-            }
+      return Fliplet.Session.get()
+        .then(async function onSessionRetrieved(session) {
+          if (session && session.entries) {
```

Comment:
Use early exits to prevent multiple indentation.

```
if (!session || !session.entries || !session.entries.dataSource) {
  if (isPreview) {
    return Fliplet.UI.Toast(userNotLoggedMessage);
  }
}

let user = session.entries.dataSource.data;
// ...

if (!sess